docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,187 @@
+that [Workers](/workers) with different Workflow & Activity Definitions operating on the same [TaskQueue](/tasks/#task-queue) do not attempt to process [Workflow Tasks](/tasks/#workflow-task) and
+[Activity Tasks](/tasks/#actvitiy-task) that they can't successfully process, according to sets of
+versions associated with that Task Queue that you define.
+
+You accomplish this goal by assigning a build identifier (a free-form string) to the code that
+defines a Worker, and specifying which build identifiers are compatible with each other by updating
+the version sets.
+
+### When and for what should you use Worker Versioning?
+
+The primary use case for this feature is the deployment of incompatible changes to short-lived
+[Workflows](/workers). On Task Queues using this feature, the Workflow starter does not need to know
+when new versions are introduced. New [Workflow Executions](/workflows#workflow-execution) run using
+the new code in the newly deployed Workers, and old (but still open) Workflow Executions are
+processed only by Workers with an appropriate version. Old workers can be decommissioned after the
+completion of any open workflows using their version.
+
+For example, if you have a Workflow that never takes longer than a day to execute, a great strategy
+is to always assign a new build identifier to every new build of your Worker and add it to the
+version sets as the new overall default. Because you know that your Workflow never takes more than a
+day to complete, you know that you'll never have to leave older Workers running for more than a day
+after you deploy the new version (assuming availability). You don't need to worry about removing old
+versions from the sets. Leaving them there is harmless.
+
+You can use this technique with longer-lived Workflows as well. Just be aware that you might need to
+keep multiple versions of Workers running while open Workflows complete.
+
+You can also use the feature to prevent a buggy codepath from being taken on currently open
+Workflows. You can do this by adding a new version to an existing set and specifying it as
+_compatible_ with an existing version on which no future Workflow Tasks should execute. Because the
+new version processes existing [Event Histories](/workflows/#event-history), it is subject to the
+normal [deterministic constraints](/workflows/#deterministic-constraints), and you might need to use
+one of the [versioning APIs](/workflows/#workflow-versioning).
+
+The feature also permits you to make incompatible changes to Activity Definitions in conjunction
+with incompatible changes to Workflow Definitions that use those Activities. This works because any
+Activity scheduled by a Workfow on the same Task Queue will, by default, only be dispatched to
+Workers that have a compatible build ID with the Workflow that scheduled it. Thus, if you wish
+to change the type signature of an Activity Definition while creating a new incompatible build ID
```

Comment:
While mentioned above, I'd stress the fact that this requires that those activities are called only from workflows on the same task queue.